pkg/dmaasbackup/periodic_backup.go

```diff
@@ -108,3 +111,67 @@ func getNextDue(cr cron.Schedule, schedule *v1alpha1.VeleroScheduleDetails, now
 	nextSync := cr.Next(lastSync)
 	return now.After(nextSync), nextSync.Sub(now)
 }
+
+func (d *dmaasBackup) cleanupPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) error {
+	d.logger.Debug("Processing cleanup for schedule")
+
+	if dbkp.Spec.PeriodicFullBackupCfg.FullBackupRetentionThreshold == 0 {
+		d.logger.Debug("Skipping cleanup since FullBackupRetentionThreshold is 0")
+		return nil
+	}
+
+	// delete backups for schedule as per fullBackupRetentionThreshold
+	// we need to retain the backups created by current active schedule and
+	// last 'FullBackupRetentionThreshold' number of schedules.
+	requiredSchedule := dbkp.Spec.PeriodicFullBackupCfg.FullBackupRetentionThreshold + 1
+
+	// scheduleCount = deleted schedules + active schedule(=1)
+	scheduleCount := getDeletedScheduleCount(dbkp) + 1
+
+	if scheduleCount < requiredSchedule {
+		d.logger.Debugf("Number of schedules are %v, required %v schedules to trigger cleanup",
+			len(dbkp.Status.VeleroSchedules),
+			requiredSchedule)
+		return nil
+	}
+
+	for index, schedule := range dbkp.Status.VeleroSchedules[requiredSchedule:] {
```

Comment:
going through the entire list from here may not be good idea. There can be a VeleroSchedule entry which is good.
Its better to just erase the last one by checking its status as Deleted and leave the deletion of any other routines for next iteration.